### PR TITLE
Docs/readme rewrite

### PR DESCRIPTION
**Pull request by alexthomas93**
# Description

README re-write.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [X] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [X] Documentation has been updated
- [ ] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [X] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,220 +99,234 @@ kg_builder = SimpleKGPipeline(
...
 
-Once the services are running, execute the following command to run the e2e tests.
+Once all the services are running, execute the following command to run the e2e tests:
 
 ```bash
 poetry run pytest tests/e2e
 ```
 
-## Further information
+## ℹ️ Additional Information
```
Comment (stellasia) [inline README.md:329]: Just wondering if we should mention ruff in the developer section?

**Review by stellasia — APPROVED** [review]
Very nice, I love the emojis :)

File: README.md (hunk 1 above)
Comment (alexthomas93) [reply README.md:329]: Good point. I'll add a little bit about pre-commit hooks too.

File: README.md [hunk 2]
```diff
@@ -1,46 +1,55 @@
...
-Follow installation instructions [here](https://pygraphviz.github.io/documentation/stable/install.html).
+Installation instructions can be found [here](https://pygraphviz.github.io/documentation/stable/install.html).
 
-## Examples
+## 💻 Example Usage
 
-### Knowledge graph construction
+The scripts below demonstrate how to get started with the package and make use of its key features.
+To run these examples, ensure that you have a Neo4j instance up and running and update the `NEO4J_URI`, `NEO4J_USERNAME`, and `NEO4J_PASSWORD` variables in each script with the details of your Neo4j instance.
+Also make sure to export your Open AI key as an environment variable named `OPENAI_API_KEY`.
```
Comment (willtai) [inline README.md:39]: Maybe highlight this is only required for the examples?

```suggestion
For the examples, make sure to export your OpenAI key as an environment variable named `OPENAI_API_KEY`.
```

**Review by willtai — APPROVED** [review]
Nice work 🧑‍🍳

**Comment by alexthomas93** [comment]
@CodiumAI-Agent /update_changelog

**Pull request merged**